packages/client-api-schema/src/jsonrpc-header-types.ts

```diff
@@ -95,18 +110,31 @@ export interface JsonRpcNotification<NotificationName = string, NotificationPara
 }
 
 /**
- * Specifies error headers as per {@link https://www.jsonrpc.org/specification | JSON-RPC 2.0 Specification }
+ * Type guard for {@link JsonRpcRequest | JsonRpcRequest}
+ *
+ * @returns true if the message is a JSON-RPC request, false otherwise
  * @beta
  */
-export type JsonRpcErrorResponse = ErrorResponse;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isJsonRpcRequest(message: any): message is JsonRpcRequest {
```

Comment:
If this really is `any` maybe we should check for `undefined`? Unless we can expect it is at least `object`